src/main/cpp/src/utilities.cu

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/types.hpp>
+#include <cudf/utilities/error.hpp>
+#include <cudf/utilities/span.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_uvector.hpp>
+#include <rmm/device_vector.hpp>
+#include <rmm/exec_policy.hpp>
+#include <rmm/mr/device/device_memory_resource.hpp>
+
+namespace spark_rapids_jni {
+
+std::unique_ptr<rmm::device_buffer> bitmask_bitwise_or(
+  std::vector<cudf::device_span<cudf::bitmask_type const>> const& input,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr)
+{
+  CUDF_EXPECTS(input.size() > 0, "Empty input");
+  auto const mask_size = (*input.begin()).size();
+  CUDF_EXPECTS(
+    std::all_of(
+      input.begin(), input.end(), [mask_size](auto mask) { return mask.size() == mask_size; }),
+    "Encountered size mismatch in inputs");
+  if (mask_size == 0) {
+    return std::make_unique<rmm::device_buffer>(rmm::device_buffer{0, stream, mr});
+  }
+
+  // move the pointers to the gpu
+  std::vector<cudf::bitmask_type const*> h_input(input.size());
+  std::transform(
+    input.begin(), input.end(), h_input.begin(), [](auto mask) { return mask.data(); });
+  rmm::device_uvector<cudf::bitmask_type const*> d_input(
+    h_input.size(), stream, rmm::mr::get_current_device_resource());
+  cudaMemcpyAsync(d_input.data(),
```

Comment:
Wrap it in `CUDF_CUDA_TRY`.

Reply:
Actually we can use `make_device_uvector_async` to cover this instead of manual memcpy.

Reply:
I'll make the change, but the counterpoint here is that `make_device_uvector_async` is a cudf detail function which is something we should generally avoid trying to call.